### PR TITLE
Add trim method to PaddedOutputString<N>

### PR DESCRIPTION
**Pull request by joakim-hove**
Small utility function used in the Well / Group POD classes.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
>It breaks the fundamental promise of the class, that the value always has N elements of character data.

Observe that the method did not mutate the state of the class - it returned a trimmed copy - `trimmed_copy()` would have been a better name - but that was so long. 

Anyway - I agree it was ugly; completely reworked in take two.

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/parser/eclipse/Utility/String.hpp [hunk 1]
```diff
@@ -19,6 +20,18 @@ typename std::decay< T >::type uppercase( T&& x ) {
...
+template<typename T>
+std::string trim_copy(const T& s) {
+    const auto c_str = s.c_str();
+    const auto end_pos = std::strchr(c_str, ' ');
+    if (end_pos == nullptr)
+        return std::string{ c_str };
+    else {
+        const std::string::size_type length = end_pos - c_str;
+        return std::string{ c_str, length };
+    }
```
Comment (bska) [inline opm/parser/eclipse/Utility/String.hpp]: If `T` here is supposed to be something that has a `c_str()`, then you could really implement this as
```C++
template <typename StringType>
std::string trim_copy(const StringType& input)
{
    auto ret = std::string(input.c_str());

    const auto end = ret.find_last_not_of(" \t\n\r\f\v");
    if (end == std::string::npos)
        return ret;
    else if (end == 0)
        return "";

    return ret.substr(0, end);
}
```

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Variant without `std::strchr` implemented - merging now.

**Pull request merged**